File: library/src/main/java/ir/vasl/library/model/Token.java
package ir.vasl.library.model;

import org.json.JSONException;
import org.json.JSONObject;

public class Token {

    public Long expires_in;
    public String token_type;
    public String refresh_token;
    public String access_token;
    public String scope;

    public Token(String tokenResult){
        JSONObject result = ParseStringToJsonObject(tokenResult);
        access_token      = ParseJsonObjectToString("access_token",result);
        token_type        = ParseJsonObjectToString("token_type",result);
        refresh_token     = ParseJsonObjectToString("refresh_token",result);
        scope             = ParseJsonObjectToString("scope",result);
        expires_in        = ParseJsonObjectToLong("expires_in",result);
    }

    private String ParseJsonObjectToString(String key, JSONObject object) {
        try {
            if (object.isNull(key))
                return "";
            return object.getString(key);
        } catch (JSONException ex) {
            return "";
        } catch (Exception ex) {
            return "";
        }
    }

    private Long ParseJsonObjectToLong(String key, JSONObject value) {
        try {
            if (value.isNull(key))
                return Long.getLong("0");
            return value.getLong(key);
        } catch (JSONException ex) {
            return Long.getLong("0");
        } catch (Exception ex) {
            return Long.getLong("0");
        }
    }

    private JSONObject ParseStringToJsonObject(String value) {
        try {
            return new JSONObject(value);
        } catch (JSONException ex) {
            return null;
        } catch (Exception ex) {
            return null;
        }
    }
}

File: library/src/main/java/com/github/library/model/Token.kt
package com.github.library.model

import org.json.JSONException
import org.json.JSONObject

class Token(tokenResult: String) {

    var expires_in: Long? = null
    var token_type: String
    var refresh_token: String
    var access_token: String
    var scope: String

    init {
        val result = parseStringToJsonObject(tokenResult)
        access_token    = parseJsonObjectToString("access_token", result)
        token_type      = parseJsonObjectToString("token_type", result)
        refresh_token   = parseJsonObjectToString("refresh_token", result)
        scope           = parseJsonObjectToString("scope", result)
        expires_in      = parseJsonObjectToLong("expires_in", result)
    }

    private fun parseJsonObjectToString(key: String, value: JSONObject?): String {
        return try {
            if (value!!.isNull(key)) "" else value.getString(key)
        } catch (ex: JSONException) {
            ""
        } catch (ex: Exception) {
            ""
        }

    }

    private fun parseJsonObjectToLong(key: String, value: JSONObject?): Long {
        return try {
            if (value!!.isNull(key)) 0.toLong() else value.getLong(key)
        } catch (ex: JSONException) {
            0.toLong()
        } catch (ex: Exception) {
            0.toLong()
        }

    }

    private fun parseStringToJsonObject(value: String): JSONObject? {
        return try {
            JSONObject(value)
        } catch (ex: JSONException) {
            null
        } catch (ex: Exception) {
            null
        }

    }
}
